### PR TITLE
HW-51430-create-addTransferMethodScreen-ui-tests

### PR DESCRIPTION
**Pull request by bhordiienko-epam**
moved  the branch from the old repo, added changes according to comments

### CODE REVIEW

File: UITests/TransferMethod/ListTransferMethodTests.swift [hunk 1]
```diff
@@ -29,17 +28,7 @@ class ListTransferMethodTests: XCTestCase {
     }
```
Comment (ekan-hw) [inline UITests/TransferMethod/ListTransferMethodTests.swift]: since it is public repo, thus we should replace the mock token instead of the real one

Comment (bhordiienko-epam) [reply UITests/TransferMethod/ListTransferMethodTests.swift]: replaced

File: UITests/TransferMethod/ListTransferMethodTests.swift [hunk 2]
```diff
@@ -253,4 +242,75 @@ class ListTransferMethodTests: XCTestCase {
...
+
+    func testListTransferMethods_verifyAfterRelaunch() {
+        setUpStandardListTransferMethod()
+        validatetestListTransferMethodsScreen()
+        XCUIDevice.shared.clickHomeAndRelaunch(app: app)
+        setUpStandardListTransferMethod()
+        validatetestListTransferMethodsScreen()
+    }
+
+    func testListTransferMethods_verifyRotateScreen() {
```
[the remark this answers is not in the document]
Comment (bhordiienko-epam) [reply UITests/TransferMethod/ListTransferMethodTests.swift]: renamed, thanks

File: UITests/TransferMethod/ListTransferMethodTests.swift [hunk 3]
```diff
@@ -253,4 +242,75 @@ class ListTransferMethodTests: XCTestCase {
...
+    func testListTransferMethods_verifyPressBackButton() {
+        setUpStandardListTransferMethod()
+        listTransferMethod.clickBackButton()
+        XCTAssertTrue(app.navigationBars["Account Settings"].exists)
+    }
+
+    func validatetestListTransferMethodsScreen() {
+        let expectedCellsCount = 4
+        let isValid = listTransferMethod.navigationBar.exists &&
+            (app.tables.element(boundBy: 0).cells.count == expectedCellsCount) &&
```
[the remark this answers is not in the document]
Comment (bhordiienko-epam) [reply UITests/TransferMethod/ListTransferMethodTests.swift]: > Create separate assertions for these please

updated, thanks

File: UITests/TransferMethod/AddTransferMethodBankAccountTests.swift [hunk 4]
```diff
@@ -108,4 +75,90 @@ class AddTransferMethodTests: XCTestCase {
...
+        XCTAssertTrue(app.keyboards.element.exists)
+        XCTAssertTrue(addTransferMethod.branchIdInput.exists)
+        XCTAssertTrue(addTransferMethod.accountNumberInput.exists)
+        XCTAssertTrue(addTransferMethod.accountTypeSelect.exists)
+        XCTAssertTrue(addTransferMethod.addTMTableView.staticTexts["Routing Number [021000022] is not valid. " +
+            "Please modify Routing Number to a valid ACH Routing Number of the branch of your bank."].exists)
+    }
+
+    private func setUpBankAccountScreen() {
+        mockServer.setupStub(url: "/rest/v3/users/usr-token/authentication-token",
```
[the remark this answers is not in the document]
Comment (bhordiienko-epam) [reply UITests/TransferMethod/AddTransferMethodBankAccountTests.swift]: > You can remove authentication-token stub and also the app launch call part as it is in the base setup function already

removed, thanks

File: UITests/TransferMethod/ListTransferMethodTests.swift [hunk 5]
```diff
@@ -253,4 +242,75 @@ class ListTransferMethodTests: XCTestCase {
...
+        validatetestListTransferMethodsScreen()
+    }
+
+    func testListTransferMethods_verifyPressBackButton() {
+        setUpStandardListTransferMethod()
+        listTransferMethod.clickBackButton()
+        XCTAssertTrue(app.navigationBars["Account Settings"].exists)
+    }
+
+    func validatetestListTransferMethodsScreen() {
```
[the remark this answers is not in the document]
Comment (bhordiienko-epam) [reply UITests/TransferMethod/ListTransferMethodTests.swift]: > Rename to just validateListTransferMethodScreen please

renamed

File: UITests/TransferMethod/AddTransferMethodBankAccountTests.swift [hunk 6]
```diff
@@ -108,4 +75,90 @@ class AddTransferMethodTests: XCTestCase {
...
+        XCUIDevice.shared.sendToBackground(app: app)
+        valideteAddTransferMethodBankAccountScreen()
+    }
+
+    func testAddTransferMethod_verifyPressBackButton() {
+        addTransferMethod.clickBackButton()
+        XCTAssertTrue(selectTransferMethodType.navigationBar.exists)
+    }
+
+    func valideteAddTransferMethodBankAccountScreen() {
```
[the remark this answers is not in the document]
Comment (bhordiienko-epam) [reply UITests/TransferMethod/AddTransferMethodBankAccountTests.swift]: > Typo with validate

fixed, thanks

File: UITests/TransferMethod/SelectTransferMethodTypeTests.swift [hunk 7]
```diff
@@ -28,7 +28,7 @@ class SelectTransferMethodTypeTests: BaseTests {
 
     private func validateSelectTransferMethodScreen() {
         var bankAccountDescription: String
-        if #available(iOS 12.0, *) {
+        if #available(iOS 11.4, *) {
```
Comment (ekan-hw) [inline UITests/TransferMethod/SelectTransferMethodTypeTests.swift]: is this "\n" also affects for the iOS below 11.4 ?

**Pull request merged**